pkg/logs/epforwarder/epforwarder.go

```diff
@@ -0,0 +1,158 @@
+		}
+	}()
+	configKeys := config.LogsConfigKeys{
+		CompressionLevel:        endpointsConfigPrefix + ".compression_level",
+		ConnectionResetInterval: endpointsConfigPrefix + ".connection_reset_interval",
+		LogsDDURL:               endpointsConfigPrefix + ".logs_dd_url",
+		DDURL:                   endpointsConfigPrefix + ".dd_url",
+		DevModeNoSSL:            endpointsConfigPrefix + ".dev_mode_no_ssl",
+		AdditionalEndpoints:     endpointsConfigPrefix + ".additional_endpoints",
+		BatchWait:               endpointsConfigPrefix + ".batch_wait",
+		BatchMaxConcurrentSend:  endpointsConfigPrefix + ".batch_max_concurrent_send",
+	}
+	endpoints, err := config.BuildHTTPEndpointsWithConfig(configKeys, hostnameEndpointPrefix)
+	if err != nil {
+		return nil, err
+	}
+	if !endpoints.UseHTTP {
+		return nil, fmt.Errorf("endpoints must be http")
+	}
+	// since some of these pipelines can be potentially very high throughput we increase the default batch send concurrency
+	// to ensure they are able to support 1000s of events per second
+	if endpoints.BatchMaxConcurrentSend <= 0 {
+		endpoints.BatchMaxConcurrentSend = 10
+	}
+	main := http.NewDestination(endpoints.Main, http.JSONContentType, destinationsContext, endpoints.BatchMaxConcurrentSend)
+	additionals := []client.Destination{}
+	for _, endpoint := range endpoints.Additionals {
+		additionals = append(additionals, http.NewDestination(endpoint, http.JSONContentType, destinationsContext, endpoints.BatchMaxConcurrentSend))
+	}
+	destinations := client.NewDestinations(main, additionals)
+	inputChan := make(chan *message.Message, 100)
+	strategy := sender.NewBatchStrategy(sender.ArraySerializer, endpoints.BatchWait, endpoints.BatchMaxConcurrentSend)
+	a := auditor.NewNullAuditor()
+	log.Debugf("Initialized event platform forwarder pipeline. eventType=%s mainHost=%s additionalHosts=%s batch_max_concurrent_send=%d", eventType, endpoints.Main.Host, joinHosts(endpoints.Additionals), endpoints.BatchMaxConcurrentSend)
+	return &passthroughPipeline{
+		sender:  sender.NewSender(inputChan, a.Channel(), destinations, strategy),
+		in:      inputChan,
+		auditor: a,
+	}, nil
+}
+
+func (p *passthroughPipeline) Start() {
+	p.auditor.Start()
+	p.sender.Start()
+}
+
+func (p *passthroughPipeline) Stop() {
+	p.sender.Stop()
+	p.auditor.Stop()
+}
+
+func joinHosts(endpoints []config.Endpoint) string {
+	var additionalHosts []string
+	for _, e := range endpoints {
+		additionalHosts = append(additionalHosts, e.Host)
+	}
+	return strings.Join(additionalHosts, ",")
+}
+
+// NewEventPlatformForwarder creates a new EventPlatformForwarder
+func NewEventPlatformForwarder() EventPlatformForwarder {
+	destinationsCtx := client.NewDestinationsContext()
+	destinationsCtx.Start()
+	pipelines := make(map[string]*passthroughPipeline)
+
+	if p, err := newHTTPPassthroughPipeline(eventTypeDBMSamples, "database_monitoring.samples", "dbquery-http-intake.logs.", destinationsCtx); err == nil {
```

Comment:
Not a big deal but these errors are silent; may be nicer to output them for future devs?

Reply:
They were actually being logged but inside `newHTTPPassthroughPipeline` which I see now is confusing. 

I refactored to make the logging more explicit https://github.com/DataDog/datadog-agent/pull/7735/commits/e9b6092664aba8e960031e32ac699ffabb7242af

Reply:
👌